mitodl/ocw-studio
# File uploads API for WebsiteContent

#### Pre-Flight checklist

#### What are the relevant tickets?
Related to #92 

#### What's this PR do?
- Installs `django-storages`
- Adds a `FileField` to the `WebsiteContent` model
- Adds the file field to the `WebsiteContent` serializers

#### How should this be manually tested?
-Go to `http://localhost:8043/api/websites/<WEBSITE_NAME>/content/`
-Use the form at the bottom to create a new object of type `resource` and choose a file to upload.
-The response should contain a link to the uploaded file, click on it and make sure the file loads.
-Set `OCW_STUDIO_USE_S3=True`, copy `AWS_` settings from CI, and restart docker containers
-Upload a file again.  This time the link should be an S3 URL.  Currently you won't have access to that URL but you can confirm the file exists in a shell:

```python
import boto3
from django.conf import settings
s3 = boto3.resource("s3") 
obj = s3.Object(settings.AWS_STORAGE_BUCKET_NAME, key="<WEBSITE_NAME>/<CONTENT_OBJECT_UUID>_<FILENAME>"
print(obj.content_type, obj.last_modified, obj.content_length)
```
    
#### Any background context you want to provide?
Some things which this PR doesn't address:
  - File type validation (not sure what if any file type restrictions should be in place)
  - Syncing uploads to OCW preview/live buckets
  - Viewing the file when `OCW_STUDIO_USE_S3=True` (The S3 URLs are currently not accessible).
  - Importing legacy OCW files
